Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2502,91 +2584,14 @@ AttributedString editingAttributedString(const SimpleRange& range, OptionSet<Inc
-        else
-            [attrs removeObjectForKey:NSStrikethroughStyleAttributeName];
-
-        if (auto ctFont = style.fontCascade().primaryFont().getCTFont())
-            [attrs setObject:(__bridge PlatformFont *)ctFont forKey:NSFontAttributeName];
-        else {
-            auto size = style.fontCascade().primaryFont().platformData().size();
-#if PLATFORM(IOS_FAMILY)
-            PlatformFont *platformFont = [PlatformFontClass systemFontOfSize:size];
-#else
-            PlatformFont *platformFont = [fontManager convertFont:WebDefaultFont() toSize:size];
-#endif
-            [attrs setObject:platformFont forKey:NSFontAttributeName];
-        }
-
-        auto textAlignment = NSTextAlignmentNatural;
-        switch (style.textAlign()) {
-        case TextAlignMode::Right:
-        case TextAlignMode::WebKitRight:
-            textAlignment = NSTextAlignmentRight;
-            break;
-        case TextAlignMode::Left:
-        case TextAlignMode::WebKitLeft:
-            textAlignment = NSTextAlignmentLeft;
-            break;
-        case TextAlignMode::Center:
-        case TextAlignMode::WebKitCenter:
-            textAlignment = NSTextAlignmentCenter;
-            break;
-        case TextAlignMode::Justify:
-            textAlignment = NSTextAlignmentJustified;
-            break;
-        case TextAlignMode::Start:
-            if (style.hasExplicitlySetDirection())
-                textAlignment = style.isLeftToRightDirection() ? NSTextAlignmentLeft : NSTextAlignmentRight;
-            break;
-        case TextAlignMode::End:
-            textAlignment = style.isLeftToRightDirection() ? NSTextAlignmentRight : NSTextAlignmentLeft;
-            break;
-        default:
-            ASSERT_NOT_REACHED();
-            break;
-        }
-
-        if (textAlignment != NSTextAlignmentNatural) {
-            auto paragraphStyle = adoptNS([PlatformNSParagraphStyle defaultParagraphStyle].mutableCopy);
-            [paragraphStyle setAlignment:textAlignment];
-            [attrs setObject:paragraphStyle.get() forKey:NSParagraphStyleAttributeName];
-        }
 
-        Color foregroundColor = style.visitedDependentColorWithColorFilter(CSSPropertyColor);
-        if (foregroundColor.isVisible())
-            [attrs setObject:cocoaColor(foregroundColor).get() forKey:NSForegroundColorAttributeName];
-        else
-            [attrs removeObjectForKey:NSForegroundColorAttributeName];
-
-        Color backgroundColor = style.visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
-        if (backgroundColor.isVisible())
-            [attrs setObject:cocoaColor(backgroundColor).get() forKey:NSBackgroundColorAttributeName];
-        else
-            [attrs removeObjectForKey:NSBackgroundColorAttributeName];
+        if (renderer)
+            setAttributesWithStyle(node, renderer->style(), includedElements, elementQualifiesForWritingToolsPreservationCache, attrs.get());
```

Comment:
Nit: Maybe `updateAttributesForStyle`? Since the attributes are carried over across iterations.